signature/policy_config.go

```diff
@@ -174,7 +172,7 @@ func (m *policyTransportScopesWithTransport) UnmarshalJSON(data []byte) error {
 		if _, ok := tmpMap[key]; ok {
 			return nil
 		}
-		if key != "" {
+		if key != "" && m.transport != nil {
```

Comment:
And another test: 

For `requirementsForImageRef`, add a test that when a transport is not in `KnownTransports` when parsing the policy, but it is in `KnownTransports` at the time of `requirementsForImageRef`, the correct set of scopes is used (more or less re-run the existing test, except creating the policy in a different way).